### PR TITLE
feat(basic-auth): basic-auth middleware

### LINKED ISSUE #26: Basic Auth

### PR DESCRIPTION
**Pull request by metrue**
Close #26

### CODE REVIEW

File: example/basic/index.js [hunk 1]
```diff
@@ -6,6 +6,13 @@ const app = new Hono()
 // Mount Builtin Middleware
 app.use('*', Middleware.poweredBy())
 app.use('*', Middleware.logger())
+app.use(
+  '*',
+  Middleware.basicAuth({
```
Comment (yusukebe) [inline example/basic/index.js:11]: I think it's good to use more general words about `name` and `pass`.
Such as *username*/*password*.

**Review by yusukebe — COMMENTED** [review]
Thank you for your PR. I have no complaints about implementation.
Commented small things about example script.

File: example/basic/index.js [hunk 2]
```diff
@@ -6,6 +6,13 @@ const app = new Hono()
 // Mount Builtin Middleware
 app.use('*', Middleware.poweredBy())
 app.use('*', Middleware.logger())
+app.use(
+  '*',
```
[the remark this answers is not in the document]
Comment (metrue) [reply example/basic/index.js:10]: Yeah, you're right, let me update it now.

**Comment by metrue** [comment]
@yusukebe 

Let me clean up the no need changes in this PR first, will let you know when it's ready for review.

**Comment by metrue** [comment]
@yusukebe It's ready for review now.

**Review by yusukebe — APPROVED** [review]
Great!

**Pull request merged**